js/names.js

```diff
@@ -0,0 +1,24 @@
+function describeOriginalName(originalNameCombination) {
+  var firstName = "";
+  if (
+    originalNameCombination === null ||
+    originalNameCombination === ""
+  ) {
+    firstName = "Name is as originally described.";
+  } else {
+    firstName =
+      "<b>Original name as described:</b> "  +
+      "<i>" +
+      originalNameCombination.split(/[_ ]/).slice(0,2).join(" ") +
```

Comment:
This is incorrect if the original name combination contains more than two words, e.g. if it was described as a subspecies or if there is a subgeneric name in the original combination.

More broadly, I think the problem is that this code relies on the MDD data following a particular format. If that format changes—for example, if we remove underscores from the original name combination field, or if we rename a column—the code will break, and a unit test can't protect against that.

Reply:
Yes, I agree that this is very targeted fix and test. 

Rather than trying to fix all the issue, I prefer to take an iterative, test driven approach to make targeted and controlled changes.

Thanks for having a look and do with it as you desire. I hope this example inspires some discussion around the realities of existing code as well as limited developer resources.